### PR TITLE
deps: upgrade github.com/golang/protobuf to v1.5.4 for better compati…

### PR DESCRIPTION
**Pull request by machine424**
…bilty, see https://github.com/golang/protobuf/issues/1596#issuecomment-1981208282

upgrade go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp to v0.49.0 to address CVE-2023-45142 even though prometheus-adapter isn't using it directly and isn't exposing any traces.

This supersedes https://github.com/kubernetes-sigs/prometheus-adapter/pull/648

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -75,32 +75,33 @@ require (
...
-	go.opentelemetry.io/otel/exporters/otlp/internal/retry v1.10.0 // indirect
-	go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.10.0 // indirect
-	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.10.0 // indirect
-	go.opentelemetry.io/otel/metric v0.31.0 // indirect
-	go.opentelemetry.io/otel/sdk v1.10.0 // indirect
-	go.opentelemetry.io/otel/trace v1.10.0 // indirect
-	go.opentelemetry.io/proto/otlp v0.19.0 // indirect
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.49.0 // indirect
+	go.opentelemetry.io/otel v1.24.0 // indirect
+	go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.17.0 // indirect
```
Comment (machine424) [inline go.mod:80]: needed to upgrade this as well to fix https://github.com/open-telemetry/opentelemetry-go/issues/4586

**Comment by machine424** [comment]
@dgrisonnet I'd like to get this merged (upstream) before merging the equivalent downstream.

**Review by dgrisonnet — APPROVED** [review]
/lgtm
/approve

**Pull request merged**